### PR TITLE
Moar Run Scanner fixes

### PR DESCRIPTION
**Pull request by apmasell**

### CODE REVIEW

File: runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultPacBio.java [hunk 1]
```diff
@@ -213,6 +214,13 @@ protected StatusResponse getStatus(String url) {
     return new RestTemplate().getForObject(url, StatusResponse.class);
   }
 
+  protected String getSampleSheet(String url) {
+    return new RestTemplate().getForObject(url, String.class);
+  }
+
+  private final static Pattern LINES = Pattern.compile("\\r?\\n");
```
Comment (sqwci) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultPacBio.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultPacBio.java [hunk 2]
```diff
@@ -213,6 +214,13 @@ protected StatusResponse getStatus(String url) {
...
+  private final static Pattern WELL_LINE = Pattern.compile("^([A-Z]\\d+),.*$");
```
Comment (sqwci) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultPacBio.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by sqwci** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DefaultPacBio.java#L178](https://github.com/TGAC/miso-lims/blob/bed623ffc837edce8f5aa49e999b1b85466fbb50/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultPacBio.java#L178): Extract the assignment out of this expression. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MINOR][MINOR] [Main.java#L28](https://github.com/TGAC/miso-lims/blob/bed623ffc837edce8f5aa49e999b1b85466fbb50/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/Main.java#L28): Remove the declaration of thrown exception 'com.fasterxml.jackson.core.JsonParseException' which is a subclass of 'java.io.IOException'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [Main.java#L28](https://github.com/TGAC/miso-lims/blob/bed623ffc837edce8f5aa49e999b1b85466fbb50/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/Main.java#L28): Remove the declaration of thrown exception 'com.fasterxml.jackson.databind.JsonMappingException' which is a subclass of 'java.io.IOException'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by callunity** [comment]
Tests failing on `testGoldens`; 
expected: `callCycle=1, imgCycle=1, numCycles=158, scoreCycle=1` 
but got `callCycle=158, imgCycle=158, numCycles=158, scoreCycle=158`
Test error or runscanner error?

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java [hunk 3]
```diff
@@ -182,6 +183,7 @@ public boolean equals(Object obj) {
     return true;
   }
```
Comment (sqwci) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java]: MAJOR Remove this unused method parameter "lane". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)

**Comment by sqwci** [comment]
SonarQube analysis reported 4 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [NotificationDto.java#L133](https://github.com/TGAC/miso-lims/blob/afcc50c22a8b03143e13d3c9f8c51ffce374ffdb/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java#L133): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [DefaultRunService.java#L421](https://github.com/TGAC/miso-lims/blob/afcc50c22a8b03143e13d3c9f8c51ffce374ffdb/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultRunService.java#L421): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [DefaultRunService.java#L421](https://github.com/TGAC/miso-lims/blob/afcc50c22a8b03143e13d3c9f8c51ffce374ffdb/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultRunService.java#L421): Remove this unused method parameter "target". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [NotificationDto.java#L134](https://github.com/TGAC/miso-lims/blob/afcc50c22a8b03143e13d3c9f8c51ffce374ffdb/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java#L134): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [DefaultRunService.java#L420](https://github.com/TGAC/miso-lims/blob/afcc50c22a8b03143e13d3c9f8c51ffce374ffdb/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultRunService.java#L420): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [DefaultRunService.java#L420](https://github.com/TGAC/miso-lims/blob/afcc50c22a8b03143e13d3c9f8c51ffce374ffdb/miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultRunService.java#L420): Remove this unused method parameter "target". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultPacBio.java [hunk 4]
```diff
@@ -174,7 +179,12 @@ private static ProcessMetadata processSampleInformation() {
       }
       Map<String, String> poolInfo = dto.getPoolNames();
       if (poolInfo == null) {
-        dto.setPoolNames(poolInfo = new HashMap<>());
+        poolInfo = new HashMap<>();
+        dto.setPoolNames(poolInfo);
+      } else if (poolInfo.containsKey(well)) {
+        // If there are multiple things assigned to this well in the sample sheet, then MISO will not be able to figure out a single pool to
+        // assign to this well. In this case, we set the pool to be the empty string so that nothing will be automatically assigned.
+        name = "";
```
Comment (djcooke) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultPacBio.java]: should probably alert or at least log so we're more likely to be aware of it happening if it does

Comment (apmasell) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultPacBio.java]: Fixed.

**Pull request merged**